### PR TITLE
 Adds sensible .gitignore files to java samples

### PR DESCRIPTION
**Pull request by etanshaul**

### CODE REVIEW

**Comment by etanshaul** [comment]
@simonz130 fyi

File: java/java-guestbook/.gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+**/.DS_Store
+.idea
```
Comment (simonz130) [inline java/java-guestbook/.gitignore]: should we include .idea? How will we add changes that trigger extension installation and so on?

File: java/java-hello-world/.gitignore [hunk 2]
```diff
@@ -1 +1,7 @@
-bin/
+**/.DS_Store
+.idea
```
Comment (simonz130) [inline java/java-hello-world/.gitignore]: ditto here

File: java/java-guestbook/.gitignore (hunk 1 above)
Comment (etanshaul) [reply java/java-guestbook/.gitignore]: yeah, you're right. fixed

**Pull request merged**